cp23/hpi-fun2/maze/executables/interactor.cpp

```diff
@@ -0,0 +1,108 @@
+#include <bits/stdc++.h>
+#include "testlib.h"
+
+using namespace std;
+
+#define WALL '#'
+#define FREE '_'
+
+void send_neighbor_info(int x, int y, vector<vector<bool>>& maze) {
+  char up = (y == 0 || maze[y - 1][x]) ? WALL : FREE;
+  char down = (y == maze.size() - 1 || maze[y + 1][x]) ? WALL : FREE;
+
+  char left = (x == 0 || maze[y][x - 1]) ? WALL : FREE;
+  char right = (x == maze.front().size() - 1 || maze[y][x + 1]) ? WALL : FREE;
+
+  cout << up  << right << down << left << endl;
+}
+
+bool is_valid_move(int x, int y, int width, int height, vector<vector<bool>>& maze) {
+  return x >= 0 && x < width && y >= 0 && y < height;
+}
+
+int main(int argc, char* argv[]) {
+  setName("describe your interactor here");
+  registerInteraction(argc, argv);
+
+  // Use inf (.in file) and ans (.ans file).
+  // Read queries from the solution using ouf, and write to it using cout.
+  // Use quit(_wa, message) or quitf(_wa, format_string, args...) to report wrong answers
+  // Exchange _wa by _pe if the format is wrong (should mostly be taken care of by the
+  // input functions of ouf).
+  // Remember that you should probably have a query limit and enforce it.
+
+  const auto width = stoi(inf.readWord());
+  const auto height = stoi(inf.readWord());
+  const auto walls = stoull(inf.readWord());
+
```

Comment:
Please use the readInt/readLong/readUnsignedLong methods. In general I'd also shy away from unsigned.